### PR TITLE
✨Scheduler: add table for user/group clusters

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- Adds a table containing available clusters (i.e. clusters table, contains name, ID, owner)
- Adds a table containing cluster to groups with their respective access rights:
  - read: means usage of cluster is possible
  - write: means modification of cluster is possible
  - delete: means deletion of cluster is possible

The postgres table clusters automatically creates an entry with the owner in the cluster_to_groups table with full rights

OPEN POINTS: the clusters table might need some additional data in order to access the cluster. TBD later

## How to test

```bash
cd packages/postgres-database
make setup-commit
```

### CODE REVIEW

**Review by mguidon — APPROVED** [review]
Looks good to me.

File: packages/postgres-database/src/simcore_postgres_database/models/clusters.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            "groups.gid",
+            name="fk_services_meta_data_gid_groups",
+            onupdate="CASCADE",
+            ondelete="RESTRICT",
+        ),
+        nullable=False,
+        doc="Identifier of the group that owns this cluster",
+    ),
+    sa.Column(
+        "created",
```
Comment (pcrespov) [inline packages/postgres-database/src/simcore_postgres_database/models/clusters.py:47]: MINOR: I thought all columns related to datetime should add ``_at`` suffix by convention, i.e. ``created_at`` and ``modified_at``

File: packages/postgres-database/tests/test_clusters.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    # cleanup
+    async with pg_engine.acquire() as conn:
+        await conn.execute(clusters.delete().where(clusters.c.id.in_(cluster_ids)))
+
+
+async def test_cluster_without_owner_forbidden(
+    create_cluster: Callable[..., Awaitable[int]]
+):
+    with pytest.raises(psycopg2.errors.NotNullViolation):
```
Comment (pcrespov) [inline packages/postgres-database/tests/test_clusters.py:68]: NOTE: see ``simcore_postgres_database.errors``. psycopg2 errors are not explicit and sometimes difficult to find. IN that module we try to collect some of the errors that we are using. I propose we use this as a indirection for all database related errors

**Review by pcrespov — APPROVED** [review]
:+1: some minor

File: packages/postgres-database/src/simcore_postgres_database/models/clusters.py (hunk 1 above)
Comment (sanderegg) [reply packages/postgres-database/src/simcore_postgres_database/models/clusters.py:47]: after discussion added an issue for this. we'll unify to something without ```_at``` as this seems to be the majority! 👍

File: packages/postgres-database/tests/test_clusters.py (hunk 2 above)
Comment (sanderegg) [reply packages/postgres-database/tests/test_clusters.py]: so if I get it right, I should indirect all the errors through ```simcore_postgres_database.errors```? Done.

**Pull request merged**